### PR TITLE
Feature/om 388

### PR DESCRIPTION
**Pull request by Citronhariom**
Descriptions : Build completed for the OM-388

### CODE REVIEW

File: force-app/main/default/classes/ManageRelatedContactFieldsTest.cls [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+           // Assertequals Logic  
+            for(contact c : [SELECT Id,Account.Subsidiary__c FROM contact WHERE Id=:contact1.id]) {             
+                if(c.Account.Subsidiary__c == 'CA'){
+                    System.assertEquals(contact1.id, user1.ContactId);
+                }else{
+                    System.assertNotEquals(contact1.id, user1.ContactId);  
+                }        
+            }
+                                   
+         test.stopTest();
```
Comment (MHigel) [inline force-app/main/default/classes/ManageRelatedContactFieldsTest.cls:72]: `Test.stopTest()` should occur after the DML, but before any query to retrieve the data being validated.

File: force-app/main/default/classes/OpportunityValidationWrapperTest.cls [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            fssetting.Customer_Success_Queue__c = testGroup.Id;
+            fssetting.Ticket_New_Status__c = 'New';
+            fssetting.PO_Required_Before_Label__c = 'PO Required Before';
+            fssetting.Opportunity_To_Order_Creation_Stage_Name__c = 'Presentation';
+            insert fssetting;
+
+            // Account creation
+            account acc = new account(Name = 'TEST_ACCOUNT', Lead_Source__c = 'Blitz Day', Phone = '1234567890', BillingCountry = 'CA',
+                BillingState = 'ON', BillingCity = 'Toronto', Subsidiary__c = 'CA', Type = 'Prospect', Account_Type__c = 'Programmer');
+            insert acc;
```
Comment (MHigel) [inline force-app/main/default/classes/OpportunityValidationWrapperTest.cls:48]: The Account's `Subsidiary__c` is assigned by using its `RecordType`.  We need to create a `List<Account>` that has Account records for each record type, then create the necessary related records for each Account.  The test should validate that the process works for Opportunity records whose `Subsidiary__c` value is "CA", and does not execute for records whose `Subsidiary__c` value is either "US" or "UK".

File: force-app/main/default/classes/OpportunityValidationWrapperTest.cls [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            system.assert(opp.Opportunity_Type__c == 'Consumables');
+            
+            // opportunitycontactRole creations
+            OpportunityContactRole ocr = new OpportunityContactRole();
+            ocr.ContactId = con.Id;
+            ocr.OpportunityId = opp.Id;
+            ocr.IsPrimary = false;
+            ocr.Role = 'Decision Maker';
+            insert ocr;           
+            system.assert(ocr.IsPrimary == false);
```
Comment (MHigel) [inline force-app/main/default/classes/OpportunityValidationWrapperTest.cls:78]: This is not a valid test.  The code being tested is not responsible for validating the setting of the `IsPrimary` field.  The test should validate the functionality being executed.

File: force-app/main/default/classes/OpportunityValidationWrapperTest.cls [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            
+            ocr.IsPrimary = true;
+            update ocr;
+
+            // Opportunity update 
+            opp.Amount = 1.0;
+            opp.StageName = 'Proposal';
+            update opp;
+           test.stopTest();
```
Comment (MHigel) [inline force-app/main/default/classes/OpportunityValidationWrapperTest.cls:87]: If the tested code fires after update, then we need to validate the output of that update.  We have to re=query for the records, and include asserts that validate the results.

**Review by MHigel — CHANGES_REQUESTED** [review]

**Comment by Citronhariom** [comment]
This Pr has multiple issue ,so created new PR - https://github.com/citron-hygiene-lp/salesforce/pull/136/files#diff-06e5d8a8cc576c55697c34d965f25210c79097419b5b7c7ce7a08921de24bcac

**Pull request closed without merging**